R/sensorreading-data.R

```diff
@@ -0,0 +1,231 @@
+    date <- strftime(dateTime[[1]][1], "%m/%d/%Y")
+    
+    #Break apart, format dates/times, put back together.
+      timeFormatting <- sapply(dateTime[[1]][2], function(s) strsplit(s,split="[-]")[[1]])
+      timeFormatting[[1]] <- sapply(timeFormatting[[1]], function(s) sub(".000","",s))
+      timeFormatting[[2]] <- paste(" (UTC",timeFormatting[[2]], ")")
+      timeFormatting <-  paste(timeFormatting[[1]],timeFormatting[[2]])
+      
+      if(!is.null(listElements$estimatedTime) && !is.na(listElements$estimatedTime)) {
+        estDateTime <- (strsplit(listElements$estimatedTime, split="[T]"))
+        estDate <- strftime(estDateTime[[1]][1], "%m/%d/%Y")
+      } else {
+        estDate <- ""
+      }
+      
+      #Get the time out of the nearest corrected iv time, don't need the date
+      if (!is.null(listElements$nearestrawTime) || is.na(listElements$nearestrawTime)) {
+        dateTimeRaw <- (strsplit(listElements$nearestrawTime, split="[T]"))
+        dateRaw <- strftime(dateTimeRaw[[1]][1], "%m/%d/%Y")
+        
+        #Break apart, format dates/times, put back together.
+        timeFormattingRaw <- sapply(dateTimeRaw[[1]][2], function(s) strsplit(s,split="[-]")[[1]])
+        timeFormattingRaw[[1]] <- sapply(timeFormattingRaw[[1]], function(s) sub(".000","",s))
+        timeFormattingRaw[[2]] <- paste(" (UTC",timeFormattingRaw[[2]], ")")
+        timeFormattingRaw <-  paste(timeFormattingRaw[[1]],timeFormattingRaw[[2]]) 
+      }
+    }
+    
+    rec <- getRecorderWithinUncertainty(listElements$recorderUncertainty, listElements$value, listElements$recorderValue)
+    ind <- getIndicatedCorrection(listElements$recorderValue, listElements$value)
+    app <- getAppliedCorrection(listElements$nearestrawValue, listElements$nearestcorrectedValue)
+    corr <- getCorrectedRef(listElements$value, listElements$nearestcorrectedValue, listElements$uncertainty)
+    
+    toAdd = c(date,
+              timeFormatting,
+              nullMask(listElements$party), 
+              nullMask(listElements$sublocation),
+              ##
+              nullMask(listElements$monitoringMethod),
+              nullMask(listElements$type),
+              nullMask(listElements$value),
+              nullMask(listElements$uncertainty),
+              ##
+              nullMask(listElements$recorderMethod),
+              nullMask(listElements$recorderType),
+              nullMask(listElements$recorderValue),
+              nullMask(listElements$recorderUncertainty), 
+              ##
+              rec, 
+              ind, 
+              app, 
+              corr,
+              ##
+              nullMask(listElements$nearestrawValue),
+              timeFormattingRaw
+              #qualifiers?
+              )
+    
+    toRet <- rbind(toRet, data.frame(t(toAdd),stringsAsFactors = FALSE))
+  }
+  colnames(toRet) <- columnNames
+  rownames(toRet) <- NULL
+  
+  return(toRet)
+}
+
+nullMask <- function(val) {
```

Comment:
@lindsaycarr  -- I carried that over from sitevisitpeak and didn't mess with a good thing. I believe it's supposed to accept all the values from the json and see if they're null and if not make the value the value, but if it is, make an empty string the value?